doc/arkode/guide/source/Usage/ARKStep_c_interface/Relaxation.rst

```diff
@@ -0,0 +1,277 @@
+   meeting the solve tolerance (determined by :c:func:`ARKStepSetRelaxResTol`
+   and :c:func:`ARKStepSetRelaxTol`), the step will be repeated with a smaller
+   step size (determined by :c:func:`ARKStepSetRelaxEtaFail`). The default
+   value is 10. Input values :math:`\leq 0` will result in the default value
+   being used.
+
+   :param arkode_mem: the ARKStep memory structure
+   :param max_iters: the maximum number of solver iterations allowed
+
+   :retval ARK_SUCCESS: the value was successfully set
+   :retval ARK_MEM_NULL: ``arkode_mem`` was ``NULL``
+   :retval ARK_RELAX_MEM_NULL: the internal relaxation memory structure was
+                               ``NULL``
+
+   .. versionadded:: 5.6.0
+
+.. c:function:: int ARKStepSetRelaxSolver(void* arkode_mem, ARKRelaxSolver solver)
+
+   Sets the nonlinear solver method used to compute the relaxation parameter.
+   The default value is ``ARK_RELAX_NEWTON``.
+
+   :param arkode_mem: the ARKStep memory structure
+   :param solver: the nonlinear solver to use: ``ARK_RELAX_BRENT``,
+                  ``ARK_RELAX_NEWTON``, or  ``ARK_RELAX_FIXEDPOINT``
+
+   :retval ARK_SUCCESS: the value was successfully set
+   :retval ARK_MEM_NULL: ``arkode_mem`` was ``NULL``
+   :retval ARK_RELAX_MEM_NULL: the internal relaxation memory structure was
+                               ``NULL``
+
+   .. versionadded:: 5.6.0
+
+.. c:function:: int ARKStepSetRelaxResTol(void* arkode_mem, sunrealtype res_tol)
+
+   Sets the nonlinear solver residual tolerance to use when computing the
+   relaxation parameter. If the residual or solution tolerance (see
+   :c:func:`ARKStepSetRelaxMaxIter`) is not reached within the maximum number of
+   iterations (determined by :c:func:`ARKStepSetRelaxMaxIters`), the step will
+   be repeated with a smaller step size (determined by
+   :c:func:`ARKStepSetRelaxEtaFail`). The default value is :math:`4 \epsilon`
+   where :math:`\epsilon` is floating-point precision. Input values
+   :math:`\leq 0.0` will result in the default value being used.
+
+   :param arkode_mem: the ARKStep memory structure
+   :param res_tol: the nonlinear solver residual tolerance to use
+
+   :retval ARK_SUCCESS: the value was successfully set
+   :retval ARK_MEM_NULL: ``arkode_mem`` was ``NULL``
+   :retval ARK_RELAX_MEM_NULL: the internal relaxation memory structure was
+                               ``NULL``
+
+   .. versionadded:: 5.6.0
+
+.. c:function:: int ARKStepSetRelaxTol(void* arkode_mem, sunrealtype rel_tol, sunrealtype abs_tol)
+
+   Sets the nonlinear solver relative and absolute tolerance on :math:`r`. If
+   the residual or solution tolerance is not reached within the maximum
+   number of iterations (determined by :c:func:`ARKStepSetRelaxMaxIters`), the
+   step will be repeated with a smaller step size (determined by
+   :c:func:`ARKStepSetRelaxEtaFail`). The default relative and absolute
+   tolerances are :math:`4 \epsilon` and :math:`10^{-14}`, respectively, where
```

Comment:
The relative tolerance is taken from what's used in Brent's method. In this case the absolute tolerance is used to check if an update is essentially zero. This could be tied to the precision as well but really it should depend on the acceptable range of values for the numbers involved so I went with a fixed value.